compiler/rustc_hir_typeck/src/generator_interior/drop_ranges/cfg_build.rs

```diff
@@ -304,8 +304,8 @@ impl<'a, 'tcx> Visitor<'tcx> for DropRangeVisitor<'a, 'tcx> {
         let mut reinit = None;
         match expr.kind {
             ExprKind::Assign(lhs, rhs, _) => {
-                self.visit_expr(lhs);
                 self.visit_expr(rhs);
+                self.visit_expr(lhs);
```

Comment:
This matches the behavior described in the Rust Reference for assignment expressions (https://doc.rust-lang.org/reference/expressions/operator-expr.html#basic-assignments) 👍